samples/bmp280-sensor-azure-iot-hub/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Device.Bmxx80;
+using Iot.Device.Common;
+using UnitsNet;
+using System.Device.Gpio;
+using Microsoft.Azure.Devices.Client;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace DNSensorAzureIoTHub
+{
+    public class Program
+    {
+        // set up IoT Hub message
+        const string DeviceID = "<replace-with-your-device-id>";
+        const string IotBrokerAddress = "<replace-with-your-iot-hub-name>.azure-devices.net";
+
+        // LED constraints 
+        const int pin = 18;
+        const int lightTime = 1000;
+        const int dimTime = 2000;
+
+        // busId number for I2C pins
+        const int busId = 1;
+
+        public static void Main()
+        {
+            Console.WriteLine($".Net IoT with BMP280 Sensor!");
+
+            // set up for LED and pin
+            using GpioController led = new();
+            led.OpenPin(pin, PinMode.Output);
```

Comment:
@Ellerbach @raffaeler @pgrawehr @joperezr interesting usage of GpioController. Feels like `Pin` class could be useful

Reply:
(I mean the naming of GpioController suggests it's treated as an led rather than for managing pins)

Reply:
Agree, we should definitely discuss this.
My first thought is to share the private `_openPins` dictionary (making it static?) so that you can control the pin either seprately or with the controller.
Probably the events logic should be revised as well.

In general, I find the ability to use single pins interesting and more intuitive for new users.

Reply:
We can have this discussion in an issue instead. The original design of the API was very much considering that we wouldn't have a Pin class, so now that we have gone through that route, a lot would have to change if we were to introduce it now.

Reply:
I just created a discussion [here](https://github.com/dotnet/iot/issues/1671) to track this.